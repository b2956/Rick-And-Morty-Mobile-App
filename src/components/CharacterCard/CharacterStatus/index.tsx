import React, { memo } from 'react';
import styled from 'styled-components/native';

interface ICharacterStatus {
    status: string, 
    race: string
}

const Wrapper = styled.View`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    margin-bottom: 10px;
`;

const StatusColor = styled.View`
    height: 7px;
    width: 7px;
    border-radius: 3.5px;
    background-color: ${props => {
        if(props.status === 'Alive')  return '#55cc44';
        if(props.status === 'Dead') return '#d63d2e';
        return 'gold';
    }};
`;

const StatusText = styled.Text`
    font-size: 13px;
    color: #fff;
    margin-left: 5px;
`;

const CharacterStatus = ({ status, race }: ICharacterStatus ) => {
    return (
        <Wrapper>
            <StatusColor status={status} />
                <StatusText>{status} - {race}</StatusText>
        </Wrapper>
    )
}

export default memo(CharacterStatus);