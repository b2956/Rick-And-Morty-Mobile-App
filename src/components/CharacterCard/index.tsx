import React, { memo } from 'react';
import styled from 'styled-components/native';

import CharacterStatus from './CharacterStatus';

export interface ICharacterProps {
    id: string
    name: string
    status: string
    species: string
    type: string
    gender: string
    origin: ILocation
    location: ILocation
    image: string
    episode: IEpisode[]
}

interface ILocation {
    name: string
    type: string
    dimension: string
}

interface IEpisode {
    id: string
    name: string
    episode: string
}

const Wrapper = styled.View`
    width: 95%;
    height: 170px;
    background-color: #606268;
    margin-bottom: 10px;
    align-self: center;
    border-radius: 10px;
    flex-direction: row;
    overflow: hidden;
`;

const CharacterImgWrapper = styled.View`
    width: 40%;
    height: 100%;
`

const CharacterImg = styled.Image`
    width: 100%;
    height: 100%;
`;

const CharacterInfoWrapper = styled.View`
    height: 100%;
    flex: 1;
    display: flex;
    flex-direction: column;
    padding-left: 10px;
`;

const CharacterName = styled.Text`
    font-size: 20px;
    color: #fff;
    font-weight: 700;
`;

const GreyText = styled.Text`
    color: rgb(158, 158, 158);
    font-size: 13px;
    font-weight: 400;
`;

const WhiteText = styled.Text`
    color: #fff;
    margin-bottom: 10px;
    font-size: 13px;
`;

const CharacterCard = ({ episode, gender, id, image, location, name, origin, species, status, type  }: ICharacterProps) => {
    return (
        <Wrapper>
            <CharacterImgWrapper>
                <CharacterImg source={{uri: image}}/>
            </CharacterImgWrapper>
            <CharacterInfoWrapper>
                <CharacterName>{name}</CharacterName>
                <CharacterStatus race={species} status={status}/>
                <GreyText>Last known location:</GreyText>
                <WhiteText>{location.name}</WhiteText>
                <GreyText>First seen in:</GreyText>
                <WhiteText>{episode[0].name}</WhiteText>
            </CharacterInfoWrapper>
        </Wrapper>
    )
}

export default memo(CharacterCard);