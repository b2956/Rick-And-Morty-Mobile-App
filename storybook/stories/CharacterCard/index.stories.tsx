import React from 'react';
import {storiesOf} from '@storybook/react-native';

import CharacterCard, {ICharacterProps} from '../../../src/components/CharacterCard';

const CharacterProps: ICharacterProps = {
    episodes: [
        {   
            id: '1',
            name: 'S01E01',
            episode: 'Pilot'
        },
        {   
            id: '2',
            name: 'S01E02',
            episode: 'Lawnmower Dog'
        }
    ],
    gender: 'male',
    id: '1',
    image: 'https://rickandmortyapi.com/api/location/64',
    location: {
        dimension: '267c',
        name: 'Earth',
        type: 'planet'
    },
    name: 'Rick',
    origin: {
        dimension: '267c',
        name: 'Earth',
        type: 'planet'
    },
    species: 'Human',
    status: 'Alive',
    type: 'Human'
};

storiesOf('CharacterCard', module)
    .add('Default', () => <CharacterCard {...CharacterProps} />);