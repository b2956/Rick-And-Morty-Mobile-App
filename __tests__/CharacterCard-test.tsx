import React from 'react';
import renderer from 'react-test-renderer';
import CharacterCard, { ICharacterProps } from '../src/components/CharacterCard';

describe('Character Card', () => {
    const CharacterProps: ICharacterProps = {
        episodes: [
            {   
                id: '1',
                name: 'S01E01',
                episode: 'Pilot'
            },
            {   
                id: '2',
                name: 'S01E02',
                episode: 'Lawnmower Dog'
            }
        ],
        gender: 'male',
        id: '1',
        image: 'https://rickandmortyapi.com/api/location/64',
        location: {
            dimension: '267c',
            name: 'Earth',
            type: 'planet'
        },
        name: 'Rick',
        origin: {
            dimension: '267c',
            name: 'Earth',
            type: 'planet'
        },
        species: 'Human',
        status: 'Alive',
        type: 'Human'
    };

    it('Should render the Character Card component correctly', () => {
        const tree = renderer.create(<CharacterCard {...CharacterProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});