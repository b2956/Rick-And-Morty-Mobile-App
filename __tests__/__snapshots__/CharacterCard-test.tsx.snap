// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Card Should render the Character Card component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#606268",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 230,
        "marginBottom": 10,
        "overflow": "hidden",
        "width": "95%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "width": "40%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://rickandmortyapi.com/api/location/64",
        }
      }
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "height": "100%",
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "700",
          },
        ]
      }
    >
      Rick
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 10,
            "width": "100%",
          },
        ]
      }
    >
      <View
        status="Alive"
        style={
          Array [
            Object {
              "backgroundColor": "#55cc44",
              "borderRadius": 3.5,
              "height": 7,
              "width": 7,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 13,
              "marginLeft": 5,
            },
          ]
        }
      >
        Alive
         - 
        Human
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(158, 158, 158)",
            "fontSize": 13,
            "fontWeight": "400",
          },
        ]
      }
    >
      Last known location:
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 13,
            "marginBottom": 10,
          },
        ]
      }
    >
      Earth
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(158, 158, 158)",
            "fontSize": 13,
            "fontWeight": "400",
          },
        ]
      }
    >
      First seen in:
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 13,
            "marginBottom": 10,
          },
        ]
      }
    >
      S01E01
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(158, 158, 158)",
            "fontSize": 13,
            "fontWeight": "400",
          },
        ]
      }
    >
      Present in Episodes:
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 13,
            "marginBottom": 10,
          },
        ]
      }
    >
      Pilot | Lawnmower Dog
    </Text>
  </View>
</View>
`;
